src/chromeDebugAdapter.ts

```diff
@@ -88,6 +89,10 @@ export class ChromeDebugAdapter extends CoreDebugAdapter {
             }
 
             if (launchUrl) {
+                // We store the launch file/url provided and temporarily launch and attach to about:blank page. Once we receive configurationDone() event, we redirect the page to this file/url
+                // This is done to facilitate hitting breakpoints on load
+                this._userRequestedUrl = launchUrl;
```

Comment:
Like @auchenberg said this should only be if breakOnLoad is enabled for 'regex'.

Reply:
Okay I was trying to solve this and it didn't work as expected as I thought and the page loaded without hitting any breakpoints. I think if we don't wait for the navigation problem can be 2 fold:

1. Chrome can actually load the page before receiving the setInstrumentationBreakpoint request at all.
2. Since in order to hit those breakpoints, we need Chrome to receive all the setBreakpoints requests anyway so that they are stored in the pendingBreakpoints array which I use to resolve the breakpoints when Chrome stops the execution on the first line.

So I think we will have to use this in both the cases for "regex" and "instrument" and not use it when it's "none"

Thoughts?

Reply:
To be fair, we can write code to basically not use pendingBreakpoints and call setBreakpoints after we stop on the scripts just for the "instrument" case but I believe it would be complicating stuff. Also in this case, I am only setting instrumentation breakpoint when setBreakpoint is called (or when user actually has a breakpoint before launch) and not in all cases. So due to this reason too, we need to have this landing page or else we need to figure out some other way to set the instrumentation point.

Reply:
Yeah it does make sense that we need to do this in both cases, either we are setting the BPs via regex or we are setting the Instrumentation BP.